Src/AutoFixtureUnitTest/DomainNameGeneratorTest.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Linq;
+using System.Text.RegularExpressions;
+using Ploeh.AutoFixture;
+using Ploeh.AutoFixture.Kernel;
+using Xunit;
+
+namespace Ploeh.AutoFixtureUnitTest
+{
+    public class DomainNameGeneratorTest
+    {
+        [Fact]
+        public void SutIsSpecimenBuilder()
+        {
+            // Fixture setup
+
+            // Exercise system
+            var sut = new DomainNameGenerator();
+            // Verify outcome
+            Assert.IsAssignableFrom<ISpecimenBuilder>(sut);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateWithNullRequestReturnsCorrectResult()
+        {
+            // Fixture setup
+            var sut = new DomainNameGenerator();
+            // Exercise system
+            var result = sut.Create(null, null);
+            // Verify outcome
+            Assert.Equal(new NoSpecimen(), result);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateWithNonDomainNameRequestReturnsCorrectResult()
+        {
+            // Fixture setup
+            var sut = new DomainNameGenerator();
+            // Exercise system
+            var result = sut.Create(typeof(object), null);
+            // Verify outcome
+            Assert.Equal(new NoSpecimen(), result);
+            // Teardown
+        }
+
+        [Fact]
+        public void CreateReturnsOneOfTheFictiousDomains()
+        {
+            // Fixture setup
+            var sut = new DomainNameGenerator();
+            // Exercise system
+            var result = sut.Create(typeof(DomainName), null);
+            // Verify outcome
+            Assert.IsAssignableFrom<DomainName>(result);
+            var actualDomainName = (DomainName)result;
```

Comment:
:+1: I didn't knew the `IsAssignableFrom` method wasn't void. (I'm still a bit unfamiliar with the `Xunit` assert methods; I use the FluentAssertions library at work, and also for my private projects.)